src/download.jl

```diff
@@ -7,11 +7,16 @@ Returns the URL of the gpkg dataset of the country
 function dataurl(dataid)
     # Only accepts GADM Dataset Provider
     if contains(dataid, "GADM")
-        provider, country_code = split(dataid, '_')
-
-        "https://biogeo.ucdavis.edu/data/gadm3.6/gpkg/gadm36_$(country_code)_gpkg.zip"
+        try
+            provider, country_code = split(dataid, '_')
+            "https://biogeo.ucdavis.edu/data/gadm3.6/gpkg/gadm36_$(country_code)_gpkg.zip"
+        catch err
+            @error "invalid dataset format, please use GADM_<country_code>."
+            throw(ArgumentError("invalid dataset format"))
+        end
```

Comment:
As explained, this can be much simpler with a regex. Are you familiar with regex?